Adding commands to quick installation

Aditya-eddy opened the pull request:
Gives feature mention in [issue](https://github.com/keploy/keploy/issues/1676)

Aditya-eddy committed:
Adding commands to quick installation

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy commented:
Hey @nehagup I have designed this for the website. Also thinking of adding different os in Available on section. WIll be waiting on feedback thanks!!
![Screenshot from 2024-03-12 12-26-15](https://github.com/keploy/website/assets/70089590/c3b1ac71-f9f0-423b-a76a-a88debc71ce8)

nehagup reviewed (changes requested):
@Aditya-eddy thanks for the PR. This requires some changes:

- The font type and color needs to match the color theme of the website, please refer the readme. 
- The cli commands also needs to match the theme, please check the other cli-command in the features section. 
- this section needs to move above the "Join the Keploy community" section. 
- You can remove the box around the section and replace the "available on" with "Start Recording Test cases with code-less integration"
- The first command can just be `curl -O https://raw.githubusercontent.com/keploy/keploy/main/keploy.sh && source` 
- second command can be `keploy record -c 'go run main.go'`
- third command can be `keploy test -c 'go run main.go' --delay 10`
- On clicking the copy-button a "success message" snackbar can be displayed
- Please add "explore more on docs" with link under the last step.
- The first step command is breaking with responsiveness

Aditya-eddy committed:
Merge branch 'main' into feature_addCommandLine

Aditya-eddy commented:
Hey @nehagup I have made the changes stated above. 
I have also  changed the color-theme. 
I have also fixed this [issue](https://github.com/keploy/keploy/issues/1681) in it.
![Screenshot from 2024-03-15 04-08-50](https://github.com/keploy/website/assets/70089590/706d4412-6862-4bb8-b317-5f6436f6ff30)

Aditya-eddy committed:
making the rest changes

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy committed:
correcting changes

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy committed:
correcting-2

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy committed:
Revert "correcting-2"

This reverts commit 08bc4c3931715866c32c71c890853a37e7626514.

Aditya-eddy committed:
Revert "correcting changes"

This reverts commit 6c570134c9b9e0c2819fc1c1b8135de1f217fb39.

Aditya-eddy committed:
Revert "making the rest changes"

This reverts commit 331c37f6d01a467d49245e0c1ea7bc4078d82a39.

Aditya-eddy committed:
Pushing changes

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy commented:
Sorry for the above mess. It has been fixed now. @nehagup

Aditya-eddy committed:
Merge branch 'main' into feature_addCommandLine

Aditya-eddy committed:
fixing errors

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Hermione2408 reviewed (changes requested):
@Aditya-eddy 

- As mentioned above this section should move above 'Community' section
- On clicking the copy-button a "success message" **snackbar** can be displayed (you can use from mui)
- Explore more on docs - should look clickable you can add a button similar to 'Documentation'
- Heading and subheading styling should be like 'Keploy for developers' section

Also please sign off DCO

Aditya-eddy commented:
I have made these changes. Does they fullfill the above requests?
![Screenshot from 2024-03-18 09-52-38](https://github.com/keploy/website/assets/70089590/d2d375cd-27b4-4913-9bb5-27a8bdda34c8)

cc - @nehagup , @Hermione2408

Aditya-eddy committed:
Adding changes

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Hermione2408 commented on app/layout.tsx:
```diff
@@ -4,7 +4,7 @@ import {Inter} from 'next/font/google'
 import Header from '@/components/ui/header'
 import React from 'react';
-
+import { AppRouterCacheProvider } from '@mui/material-nextjs/v13-appRouter';
```
Hey @Aditya-eddy I don't think it is required can't you directly use mui component in your section?

Aditya-eddy replied:
fixed it!

Aditya-eddy committed:
Adding Changes

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Hermione2408 reviewed (approved):
LGTM @nehagup

nehagup commented on components/installation.tsx:
```diff
@@ -0,0 +1,191 @@
+  const [open, setOpen] = React.useState(false);
+
+  const handleClick = () => {
+    
```
Why is this empty?

Aditya-eddy replied:
I've fixed it now!

nehagup commented on components/installation.tsx:
```diff
@@ -0,0 +1,191 @@
+            Explore more on Docs{" "}
+          </a>
+        </div>
+        <Snackbar
```
Can we make the styling same as the cli styling? will make it consistent

Aditya-eddy replied:
Done!

nehagup commented on components/installation.tsx:
```diff
@@ -0,0 +1,191 @@
+        </div>
+        <Snackbar
+          open={open}
+          autoHideDuration={6000}
```
3 sec is also fine

Aditya-eddy replied:
OKay!!

nehagup commented on components/installation.tsx:
```diff
@@ -0,0 +1,191 @@
+  const action = (
+    <React.Fragment>
+      <Button color="success" size="small" onClick={handleClose}>
+        UNDO
```
Instead of Undo we can add a close svg icon.

Aditya-eddy replied:
Okay got it!

nehagup reviewed (commented):
@aditya-eddy Thanks for the changes! A few more changes required to make it production ready. 

<img width="383" alt="Screenshot 2024-03-19 at 2 53 02 PM" src="https://github.com/keploy/website/assets/15074229/b1a30805-2c7d-4273-8ed2-4cbf74798eed">

1. Copy button for Installing command is hidden for sm screen size. 
2. Can you please change the copy button to be same as one used in the one-click CLI command in the above section?
3. To increase the content in the section, we can add the following one-liners under each step? Please refer to the attached screenshot for the Step title, one liner. Also I've switched back to the CMD_TO_RUN_APP because we've added the description to explain this now. Please adjust the paddings and styling of code commands. I've also changed the button text for more relevancy. 
  
<img width="1061" alt="Screenshot 2024-03-19 at 4 10 27 PM" src="https://github.com/keploy/website/assets/15074229/4a303dae-917e-4f3d-872a-36ff81b15722">

nehagup committed:
Merge branch 'main' into feature_addCommandLine

Aditya-eddy committed:
Merge branch 'main' into feature_addCommandLine

Aditya-eddy commented:
Hey @nehagup how does this look or should i add some more padding?
If this looks good I'll push the commit.
![Screenshot from 2024-03-20 04-25-03](https://github.com/keploy/website/assets/70089590/de01c1a8-d53e-401e-a1a0-74568083811e)

![Screenshot from 2024-03-20 04-26-22](https://github.com/keploy/website/assets/70089590/b20146c9-4081-45fe-87a5-94ada5db4071)

Aditya-eddy committed:
adding icons

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy committed:
Resolving conflicts

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Hermione2408 reviewed (changes requested):
Hey @Aditya-eddy 
![Screenshot from 2024-03-27 13-11-47](https://github.com/keploy/website/assets/78696767/ce1e875c-ca23-443c-a961-55fee47abaf8)
In the mobile view Snackbar is getting overlapped by the live chat icon , adjust the positioning to prevent this overlap. Also ensure the text on the button is centered.
Please remove the SVG that is not getting used now

Hermione2408 commented on components/installation.tsx:
```diff
@@ -0,0 +1,196 @@
+              />
+            </div>
+          </div>
+            <p className=" text-gray-600 mb-0 mt-2 text-xs md:text-sm lg:text-base">
```
keep the text-color the same as other one-liner

nehagup reviewed (commented):
@Aditya-eddy good improvement. 

- Like @Hermione2408 mentioned please shorten the width of the snackbar so that it doesn't overlap. 
- Please add code formatting without quotes to the file names or cli commands, example `CMD_TO_RUN_APP` is the command to start you app. Say, `python main.py` or `npm start` or ...
- Also on smaller screen size can you please adjust the button to fix in one liner?

Aditya-eddy commented:
Hey @nehagup how does this look. Here in the code blocks should give the a lighter theme or should give them same code blocks background as in the above code blocks
![Screenshot from 2024-03-27 17-31-03](https://github.com/keploy/website/assets/70089590/30ffa269-6e6b-4da2-8574-e161f0616703)
![Screenshot from 2024-03-27 17-30-46](https://github.com/keploy/website/assets/70089590/d21deb6a-280b-40c0-99d2-54e4b36669ce)

nehagup commented:
> Hey @nehagup how does this look. Here in the code blocks should give the a lighter theme or should give them same code blocks background as in the above code blocks

@Aditya-eddy I think it's a good idea to keep it in lighter theme, so that it blends with the background. you can try to find a shade darker than bg-color of this page in tailwind config.

Aditya-eddy commented:
Hey  @nehagup  does this look good:-
![Screenshot from 2024-03-28 08-43-49](https://github.com/keploy/website/assets/70089590/6f713f02-0e1e-4227-bf9f-42e877faa5f9)

Aditya-eddy committed:
resolving conflict

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy committed:
updating package.json

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy committed:
resolving conflicts

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy commented:
Hey @nehagup any new changes in this?

nehagup committed:
Merge branch 'main' into feature_addCommandLine

nehagup commented:
@Hermione2408 can you please review?

Hermione2408 commented:
Hey @Aditya-eddy there are conflicts in your PR please resolve them.  Also, it seems like the Installation component might have been missed in app/page.tsx during the conflict resolution. Could you check that?

Hermione2408 commented:
> @Hermione2408 can you please review?

Sure

Aditya-eddy committed:
resolving conflicts

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy committed:
adding installation

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Aditya-eddy commented:
Yeah I forgot to add after resolving conflicts fixed it now. Check please! 
cc - @Hermione2408

Hermione2408 commented on components/installation.tsx:
```diff
@@ -0,0 +1,213 @@
+          className="max-w-4xl mx-auto pb-12 md:pb-16 "
+          data-aos="zoom-y-out"
+        >
+          <h1 className="text-2xl text-secondary-300 font-semibold leading-tighter tracking-tighter mb-2 ">
```
the margin is inconsistent please change it to mb-4

Aditya-eddy replied:
hey resolved it!.

Hermione2408 reviewed (changes requested):
A minor change from my side just resolve conflicts, rest looks fine!

Aditya-eddy committed:
resolving conflicts and changes

Signed-off-by: Aditya-eddy <aditya282003@gmail.com>

Hermione2408 reviewed (commented):
LGTM @nehagup

nehagup reviewed (approved):
Thanks @Aditya-eddy for the PR! LGTM! 

There are spaces missing in Database,Redis,Kafka, but you can take that up in future PRs. :)

Aditya-eddy commented:
I could have fixed in this only :smile:
